### PR TITLE
make assember for residual

### PR DESCRIPTION
**Pull request by fredrikekre**
Such symmetry!

screenshot from 2016-12-14 17-12-23

### CODE REVIEW

File: src/assembler.jl [hunk 1]
```diff
@@ -51,3 +51,12 @@ assembled values.
...
     return sparse(a.I, a.J, a.V)
 end
+
+"""
+    assemble!(g, ge, edof)
+
+Assembles the element residual `ge` into the global residual vector `g`.
+"""
+@inline function assemble!{T}(g::AbstractVector{T}, ge::AbstractVector{T}, edof::AbstractVector{Int})
+    g[edof] += ge
```
Comment (KristofferC) [inline src/assembler.jl]: Bad performance from `+=` due to the extra array created. Use a loop :)

Comment (fredrikekre) [inline src/assembler.jl]: 25 times faster for a small test

File: src/assembler.jl [hunk 2]
```diff
@@ -33,10 +33,10 @@ end
 
 Assembles the element matrix `Ke` into `a`.
 """
-function assemble!{T}(a::Assembler{T}, Ke::AbstractMatrix{T}, edof::AbstractVector{Int})
+@inline function assemble!{T}(a::Assembler{T}, Ke::AbstractMatrix{T}, edof::AbstractVector{Int})
     n_dofs = length(edof)
     append!(a.V, Ke[:])
-    for dof1 in 1:n_dofs, dof2 in 1:n_dofs
+    @inbounds for dof1 in 1:n_dofs, dof2 in 1:n_dofs
```
Comment (KristofferC) [inline src/assembler.jl]: This is not safe if user puts a bad `edof`. Should also not be inlined

File: src/assembler.jl [hunk 3]
```diff
@@ -51,3 +51,14 @@ assembled values.
...
     return sparse(a.I, a.J, a.V)
 end
+
+"""
+    assemble!(g, ge, edof)
+
+Assembles the element residual `ge` into the global residual vector `g`.
+"""
+@inline function assemble!{T}(g::AbstractVector{T}, ge::AbstractVector{T}, edof::AbstractVector{Int})
+    @inbounds for i in 1:length(ge)
```
Comment (fredrikekre) [inline src/assembler.jl]: Do we need to `@assert length(ge) == length(edof) < length(g)` ?
We don't do such a check for the stiffness.

Comment (KristofferC) [inline src/assembler.jl]: asserts are typically used for purely internal errors (like getting into an "impossible" state), while throwing is done for user errors.

**Comment by fredrikekre** [comment]
What was the test failure here? Didn't seem related though

**Pull request merged**